server/workflow/workflow_server.go

```diff
@@ -298,6 +299,12 @@ func (s *workflowServer) DeleteWorkflow(ctx context.Context, req *workflowpkg.Wo
 	if err != nil {
 		return nil, err
 	}
+	if req.Force {
+		_, err := auth.GetWfClient(ctx).ArgoprojV1alpha1().Workflows(wf.Namespace).Patch(ctx, wf.Name, types.MergePatchType, []byte("{\"metadata\":{\"finalizers\":null}}"), metav1.PatchOptions{})
+		if err != nil {
+			return nil, err
+		}
+	}
 	err = auth.GetWfClient(ctx).ArgoprojV1alpha1().Workflows(wf.Namespace).Delete(ctx, wf.Name, metav1.DeleteOptions{PropagationPolicy: argoutil.GetDeletePropagation()})
```

Comment:
I think DeleteOptions might have force option?

Reply:
It does not, Bala suggested we do this method instead.

```
type DeleteOptions struct {
	TypeMeta `json:",inline"`

	
	GracePeriodSeconds *int64 `json:"gracePeriodSeconds,omitempty" protobuf:"varint,1,opt,name=gracePeriodSeconds"`

	Preconditions *Preconditions `json:"preconditions,omitempty" protobuf:"bytes,2,opt,name=preconditions"`

	OrphanDependents *bool `json:"orphanDependents,omitempty" protobuf:"varint,3,opt,name=orphanDependents"`

	PropagationPolicy *DeletionPropagation `json:"propagationPolicy,omitempty" protobuf:"varint,4,opt,name=propagationPolicy"`


	DryRun []string `json:"dryRun,omitempty" protobuf:"bytes,5,rep,name=dryRun"`
}
```

Reply:
OK. What does `kubectl delete --force` do? Can you run that with debug? I'd expect it be the same

Reply:
I patched this wf to add the artifact-gc finalizer and tried running force delete. The console prints that the workflow is deleted but the process doesn't actually finish and the workflow isn't deleted. Same behavior as just running `kubectl delete` or `argo delete` without removing the finalizer.

```
warning: Immediate deletion does not wait for confirmation that the running resource has been terminated. The resource may continue to run on the cluster indefinitely.
workflow.argoproj.io "hello-world-scgnt" force deleted
```

Reply:
kubectl delete --force basically make GracePeriod=0 delete the resource ASAP. kubectl delete doesn't support the `finalizer` removable. They don't want to bypass the other controllers which are holding the finalizer.